### ISSUE TITLE
Create beta of 7.5

### ISSUE THREAD

**Issue by peastman** [openmm/openmm#2766]
I think we're almost ready to build a beta of 7.5.  #2765 is the only thing left that I specifically want to get in first.  Is there anything else that should be included?

**Comment by jchodera** [comment]
I'm way behind in commenting on #2649, but coming up with a workable solution for this would be extremely valuable for FAH.
I'll comment there now.

**Comment by jchodera** [comment]
Also, not sure if you had noticed, but our omnia build infrastructure is now irreparably broken. Part of the call tomorrow is to talk through how we can accelerate the move to conda-forge. We very likely need to do that first before we can build a 7.5.0 beta.

**Comment by peastman** [comment]
As far as the code is concerned I think we're ready to build the beta.  Now it's just a question of the build infrastructure.

**Comment by jchodera** [comment]
We could use the current omnia-dev build infrastructure to build a beta for testing feature completeness. We will still have to get the build infrastructure assembled to build and test release candidates though.

**Comment by peastman** [comment]
I think that would be fine.  Betas are about testing new features, and that can happen in parallel with getting the build system into shape.

**Comment by jchodera** [comment]
Should we cut a 7.5beta1 release for this? Or just designate a git hash and call it 7.5.0 but upload to the beta label of omnia?

**Comment by peastman** [comment]
> Or just designate a git hash and call it 7.5.0 but upload to the beta label of omnia?

Let's do that to start with.  Once the beta is out we can add a tag to indicate the revision it used.

**Comment by jchodera** [comment]
Which commit hash do you want to use? Current master?

**Comment by peastman** [comment]
Yes, that would be fine.  That's cc3c4b54305fd53b535ddf26df15d9e1bb903e47.

**Comment by jaimergp** [comment]
Some variants already available as artifacts here: https://github.com/openmm/openmm-feedstock/pull/2

**Comment by jaimergp** [comment]
Beta packages from `omnia-dev-recipes` being slowly pushed to:

https://anaconda.org/omnia-dev/openmm/files?channel=beta

**Comment by jchodera** [comment]
@peastman Do you want to put up an announcement about the beta?

**Comment by peastman** [comment]
I'm working on testing the packages.  I just tried the Mac install and encountered a strange issue.  I created a new environment with Python 3.7, then told it to install with

```
conda install -c omnia-dev/label/beta openmm
```

Conda gave me a long list of packages it was going to install including

```
  python             conda-forge/osx-64::python-3.8.5-h85f3143_2_cpython
```

That is, when I tried to install OpenMM into Python 3.7, conda insisted on replacing it with Python 3.8.

I'll try the Linux packages next.

**Comment by jaimergp** [comment]
Does it resolve well if you do `conda install -c omnia-dev/label/beta openmm python=3.7`?

**Comment by peastman** [comment]
Only sort of.  If I do that, the list of packages to be installed includes

```
  python             conda-forge/osx-64::python-3.7.8-hc9dea61_1_cpython
```

So it now downloads Python 3.7 instead of Python 3.8.  But it still insists on installing a new Python rather than keeping the one I already have.

**Comment by jaimergp** [comment]
Can you post the `conda list` output of the existing environment?

**Comment by peastman** [comment]
I tried to install on Linux with the comand

```
conda install -c omnia-dev/label/beta openmm
```

It failed with these errors.

```
UnsatisfiableError: The following specifications were found
to be incompatible with the existing python installation in your environment:

Specifications:

  - openmm -> python[version='>=2.7,<2.8.0a0|>=3.5,<3.6.0a0']

Your python: python=3.7

If python is on the left-most side of the chain, that's the version you've asked for.
When python appears to the right, that indicates that the thing on the left is somehow
not available for the python version you are constrained to. Note that conda will not
change your python version to a different minor version unless you explicitly specify
that.

The following specifications were found to be incompatible with your CUDA driver:

  - feature:/linux-64::__cuda==10.2=0
  - feature:|@/linux-64::__cuda==10.2=0

Your installed CUDA driver is: 10.2
```

**Comment by peastman** [comment]
> Can you post the conda list output of the existing environment?

```
# packages in environment at /Users/peastman/miniconda3/envs/test:
#
# Name                    Version                   Build  Channel
```

That's it!  It's a new environment just created with `conda create --name test`.

**Comment by jaimergp** [comment]
That's the same problem we saw on https://github.com/openmm/openmm/issues/2683. I have to look into that. Is `10.2` your CUDA version or is that a bug too?

**Comment by peastman** [comment]
Yes, I have 10.2 installed (through the standard Nvidia installer, not through conda).

**Comment by jaimergp** [comment]
I can't reproduce this on a GPU-enabled (CUDA 10.2) HPC instance:

```
Job <16402363> is submitted to queue <gpuqueue>.
<<Waiting for dispatch ...>>
<<Starting on lu10>>
(base) bash-4.2$ nvidia-smi
Thu Aug  6 11:49:02 2020
+-----------------------------------------------------------------------------+
| NVIDIA-SMI 440.33.01    Driver Version: 440.33.01    CUDA Version: 10.2     |
|-------------------------------+----------------------+----------------------+
| GPU  Name        Persistence-M| Bus-Id        Disp.A | Volatile Uncorr. ECC |
| Fan  Temp  Perf  Pwr:Usage/Cap|         Memory-Usage | GPU-Util  Compute M. |
|===============================+======================+======================|
|   0  GeForce RTX 2080    On   | 00000000:02:00.0 Off |                  N/A |
| 29%   49C    P8    23W / 215W |      0MiB /  7982MiB |      0%   E. Process |
+-------------------------------+----------------------+----------------------+

+-----------------------------------------------------------------------------+
| Processes:                                                       GPU Memory |
|  GPU       PID   Type   Process name                             Usage      |
|=============================================================================|
|  No running processes found                                                 |
+-----------------------------------------------------------------------------+
(base) bash-4.2$ conda create -n openmm-beta -c conda-forge -c omnia-dev/label/beta openmm
Collecting package metadata (current_repodata.json): done
Solving environment: done

## Package Plan ##

  environment location: /home/rodriguj/.conda/envs/openmm-beta

  added / updated specs:
    - openmm


The following packages will be downloaded:

    package                    |            build
    ---------------------------|-----------------
    _openmp_mutex-4.5          |            1_gnu          22 KB  conda-forge
    ld_impl_linux-64-2.34      |       hc38a660_9         612 KB  conda-forge
    libgcc-ng-9.3.0            |      h24d8f2e_14         7.8 MB  conda-forge
    libgfortran-ng-7.5.0       |      hdf63c60_14         1.3 MB  conda-forge
    libgomp-9.3.0              |      h24d8f2e_14         374 KB  conda-forge
    libopenblas-0.3.10         |pthreads_hb3c22a3_4         7.8 MB  conda-forge
    libstdcxx-ng-9.3.0         |      hdf63c60_14         4.0 MB  conda-forge
    numpy-1.19.1               |   py38h8854b6b_0         5.3 MB  conda-forge
    openmm-7.5.0b1             |cc3c4b5_py38_cuda102_0        10.9 MB  omnia-dev/label/beta
    openssl-1.1.1g             |       h516909a_1         2.1 MB  conda-forge
    pip-20.2.1                 |             py_0         1.1 MB  conda-forge
    python-3.8.5               |h4d41432_2_cpython        71.0 MB  conda-forge
    setuptools-49.2.1          |   py38h32f6830_0         931 KB  conda-forge
    ------------------------------------------------------------
                                           Total:       113.1 MB

The following NEW packages will be INSTALLED:

  _libgcc_mutex      conda-forge/linux-64::_libgcc_mutex-0.1-conda_forge
  _openmp_mutex      conda-forge/linux-64::_openmp_mutex-4.5-1_gnu
  ca-certificates    conda-forge/linux-64::ca-certificates-2020.6.20-hecda079_0
  certifi            conda-forge/linux-64::certifi-2020.6.20-py38h32f6830_0
  fftw               conda-forge/linux-64::fftw-3.3.8-nompi_h7f3a6c3_1111
  ld_impl_linux-64   conda-forge/linux-64::ld_impl_linux-64-2.34-hc38a660_9
  libblas            conda-forge/linux-64::libblas-3.8.0-17_openblas
  libcblas           conda-forge/linux-64::libcblas-3.8.0-17_openblas
  libffi             conda-forge/linux-64::libffi-3.2.1-he1b5a44_1007
  libgcc-ng          conda-forge/linux-64::libgcc-ng-9.3.0-h24d8f2e_14
  libgfortran-ng     conda-forge/linux-64::libgfortran-ng-7.5.0-hdf63c60_14
  libgomp            conda-forge/linux-64::libgomp-9.3.0-h24d8f2e_14
  liblapack          conda-forge/linux-64::liblapack-3.8.0-17_openblas
  libopenblas        conda-forge/linux-64::libopenblas-0.3.10-pthreads_hb3c22a3_4
  libstdcxx-ng       conda-forge/linux-64::libstdcxx-ng-9.3.0-hdf63c60_14
  ncurses            conda-forge/linux-64::ncurses-6.2-he1b5a44_1
  numpy              conda-forge/linux-64::numpy-1.19.1-py38h8854b6b_0
  openmm             omnia-dev/label/beta/linux-64::openmm-7.5.0b1-cc3c4b5_py38_cuda102_0
  openssl            conda-forge/linux-64::openssl-1.1.1g-h516909a_1
  pip                conda-forge/noarch::pip-20.2.1-py_0
  python             conda-forge/linux-64::python-3.8.5-h4d41432_2_cpython
  python_abi         conda-forge/linux-64::python_abi-3.8-1_cp38
  readline           conda-forge/linux-64::readline-8.0-he28a2e2_2
  setuptools         conda-forge/linux-64::setuptools-49.2.1-py38h32f6830_0
  sqlite             conda-forge/linux-64::sqlite-3.32.3-hcee41ef_1
  tk                 conda-forge/linux-64::tk-8.6.10-hed695b0_0
  wheel              conda-forge/noarch::wheel-0.34.2-py_1
  xz                 conda-forge/linux-64::xz-5.2.5-h516909a_1
  zlib               conda-forge/linux-64::zlib-1.2.11-h516909a_1006


Proceed ([y]/n)? y


Downloading and Extracting Packages
libgcc-ng-9.3.0      | 7.8 MB    | ###################################################################################################################################################### | 100%
libstdcxx-ng-9.3.0   | 4.0 MB    | ###################################################################################################################################################### | 100%
ld_impl_linux-64-2.3 | 612 KB    | ###################################################################################################################################################### | 100%
_openmp_mutex-4.5    | 22 KB     | ###################################################################################################################################################### | 100%
numpy-1.19.1         | 5.3 MB    | ###################################################################################################################################################### | 100%
pip-20.2.1           | 1.1 MB    | ###################################################################################################################################################### | 100%
openssl-1.1.1g       | 2.1 MB    | ###################################################################################################################################################### | 100%
libgfortran-ng-7.5.0 | 1.3 MB    | ###################################################################################################################################################### | 100%
libgomp-9.3.0        | 374 KB    | ###################################################################################################################################################### | 100%
openmm-7.5.0b1       | 10.9 MB   | ###################################################################################################################################################### | 100%
python-3.8.5         | 71.0 MB   | ###################################################################################################################################################### | 100%
libopenblas-0.3.10   | 7.8 MB    | ###################################################################################################################################################### | 100%
setuptools-49.2.1    | 931 KB    | ###################################################################################################################################################### | 100%
Preparing transaction: done
Verifying transaction: done
Executing transaction: done
#
# To activate this environment, use
#
#     $ conda activate openmm-beta
#
# To deactivate an active environment, use
#
#     $ conda deactivate

(base) bash-4.2$ conda activate openmm-beta
(openmm-beta) bash-4.2$ python -m simtk.testInstallation

OpenMM Version: 7.5
Git Revision: cc3c4b54305fd53b535ddf26df15d9e1bb903e47

There are 4 Platforms available:

1 Reference - Successfully computed forces
2 CPU - Successfully computed forces
3 CUDA - Successfully computed forces
4 OpenCL - Successfully computed forces

Median difference in forces between platforms:

Reference vs. CPU: 6.30349e-06
Reference vs. CUDA: 6.73129e-06
CPU vs. CUDA: 7.42031e-07
Reference vs. OpenCL: 6.75312e-06
CPU vs. OpenCL: 8.10068e-07
CUDA vs. OpenCL: 2.65926e-07

All differences are within tolerance.
```

**Comment by jaimergp** [comment]
If I create an empty test env, and then install OpenMM on top, it also works:


```
(openmm-beta) bash-4.2$ conda activate test
(test) bash-4.2$ conda install -c conda-forge -c omnia-dev/label/beta openmm
Collecting package metadata (current_repodata.json): done
Solving environment: done

## Package Plan ##

  environment location: /home/rodriguj/.conda/envs/test

  added / updated specs:
    - openmm


The following NEW packages will be INSTALLED:

  _libgcc_mutex      conda-forge/linux-64::_libgcc_mutex-0.1-conda_forge
  _openmp_mutex      conda-forge/linux-64::_openmp_mutex-4.5-1_gnu
  ca-certificates    conda-forge/linux-64::ca-certificates-2020.6.20-hecda079_0
  certifi            conda-forge/linux-64::certifi-2020.6.20-py38h32f6830_0
  fftw               conda-forge/linux-64::fftw-3.3.8-nompi_h7f3a6c3_1111
  ld_impl_linux-64   conda-forge/linux-64::ld_impl_linux-64-2.34-hc38a660_9
  libblas            conda-forge/linux-64::libblas-3.8.0-17_openblas
  libcblas           conda-forge/linux-64::libcblas-3.8.0-17_openblas
  libffi             conda-forge/linux-64::libffi-3.2.1-he1b5a44_1007
  libgcc-ng          conda-forge/linux-64::libgcc-ng-9.3.0-h24d8f2e_14
  libgfortran-ng     conda-forge/linux-64::libgfortran-ng-7.5.0-hdf63c60_14
  libgomp            conda-forge/linux-64::libgomp-9.3.0-h24d8f2e_14
  liblapack          conda-forge/linux-64::liblapack-3.8.0-17_openblas
  libopenblas        conda-forge/linux-64::libopenblas-0.3.10-pthreads_hb3c22a3_4
  libstdcxx-ng       conda-forge/linux-64::libstdcxx-ng-9.3.0-hdf63c60_14
  ncurses            conda-forge/linux-64::ncurses-6.2-he1b5a44_1
  numpy              conda-forge/linux-64::numpy-1.19.1-py38h8854b6b_0
  openmm             omnia-dev/label/beta/linux-64::openmm-7.5.0b1-cc3c4b5_py38_cuda102_0
  openssl            conda-forge/linux-64::openssl-1.1.1g-h516909a_1
  pip                conda-forge/noarch::pip-20.2.1-py_0
  python             conda-forge/linux-64::python-3.8.5-h4d41432_2_cpython
  python_abi         conda-forge/linux-64::python_abi-3.8-1_cp38
  readline           conda-forge/linux-64::readline-8.0-he28a2e2_2
  setuptools         conda-forge/linux-64::setuptools-49.2.1-py38h32f6830_0
  sqlite             conda-forge/linux-64::sqlite-3.32.3-hcee41ef_1
  tk                 conda-forge/linux-64::tk-8.6.10-hed695b0_0
  wheel              conda-forge/noarch::wheel-0.34.2-py_1
  xz                 conda-forge/linux-64::xz-5.2.5-h516909a_1
  zlib               conda-forge/linux-64::zlib-1.2.11-h516909a_1006


Proceed ([y]/n)? y

Preparing transaction: done
Verifying transaction: done
Executing transaction: done
```

**Comment by jaimergp** [comment]
My `conda info` also shows `__cuda==10.2`:

```
 bash-4.2$ conda info

     active environment : test
    active env location : /home/rodriguj/.conda/envs/test
            shell level : 3
       user config file : /home/rodriguj/.condarc
 populated config files : /home/rodriguj/.condarc
          conda version : 4.8.3
    conda-build version : 3.18.11
         python version : 3.7.3.final.0
       virtual packages : __cuda=10.2
                          __glibc=2.17
       base environment : /opt/common/CentOS_7/anaconda/anaconda3  (read only)
           channel URLs : https://conda.anaconda.org/conda-forge/linux-64
                          https://conda.anaconda.org/conda-forge/noarch
                          https://repo.anaconda.com/pkgs/main/linux-64
                          https://repo.anaconda.com/pkgs/main/noarch
                          https://repo.anaconda.com/pkgs/r/linux-64
                          https://repo.anaconda.com/pkgs/r/noarch
                          https://conda.anaconda.org/omnia/linux-64
                          https://conda.anaconda.org/omnia/noarch
          package cache : /opt/common/CentOS_7/anaconda/anaconda3/pkgs
                          /home/rodriguj/.conda/pkgs
       envs directories : /home/rodriguj/.conda/envs
                          /opt/common/CentOS_7/anaconda/anaconda3/envs
               platform : linux-64
             user-agent : conda/4.8.3 requests/2.23.0 CPython/3.7.3 Linux/3.10.0-957.12.2.el7.x86_64 centos/7.6.1810 glibc/2.17
                UID:GID : 1648:4017
             netrc file : None
           offline mode : False
```

**Comment by peastman** [comment]
Interestingly, my base environment has CUDA 10.1 installed in it.  I don't know how it got there, since I'm pretty sure I didn't install it myself.  Presumably some other package brought it along.  But conda is saying I have 10.2, not 10.1, and it complains about the driver being incompatible, not the toolkit.

Here's the output of `nvidia-smi`.

```
Thu Aug  6 09:42:59 2020       
+-----------------------------------------------------------------------------+
| NVIDIA-SMI 440.44       Driver Version: 440.44       CUDA Version: 10.2     |
|-------------------------------+----------------------+----------------------+
| GPU  Name        Persistence-M| Bus-Id        Disp.A | Volatile Uncorr. ECC |
| Fan  Temp  Perf  Pwr:Usage/Cap|         Memory-Usage | GPU-Util  Compute M. |
|===============================+======================+======================|
|   0  TITAN V             Off  | 00000000:01:00.0  On |                  N/A |
| 28%   33C    P8    24W / 250W |    158MiB / 12063MiB |      1%      Default |
+-------------------------------+----------------------+----------------------+
                                                                               
+-----------------------------------------------------------------------------+
| Processes:                                                       GPU Memory |
|  GPU       PID   Type   Process name                             Usage      |
|=============================================================================|
|    0      1030      G   /usr/lib/xorg/Xorg                           132MiB |
|    0      1910      G   compiz                                        24MiB |
+-----------------------------------------------------------------------------+
```

**Comment by peastman** [comment]
And here is `conda info`.

```
     active environment : base
    active env location : /home/peastman/miniconda3
            shell level : 1
       user config file : /home/peastman/.condarc
 populated config files : 
          conda version : 4.8.3
    conda-build version : 3.18.10
         python version : 3.7.3.final.0
       virtual packages : __cuda=10.2
                          __glibc=2.23
       base environment : /home/peastman/miniconda3  (writable)
           channel URLs : https://repo.anaconda.com/pkgs/main/linux-64
                          https://repo.anaconda.com/pkgs/main/noarch
                          https://repo.anaconda.com/pkgs/r/linux-64
                          https://repo.anaconda.com/pkgs/r/noarch
          package cache : /home/peastman/miniconda3/pkgs
                          /home/peastman/.conda/pkgs
       envs directories : /home/peastman/miniconda3/envs
                          /home/peastman/.conda/envs
               platform : linux-64
             user-agent : conda/4.8.3 requests/2.22.0 CPython/3.7.3 Linux/4.15.0-112-generic ubuntu/16.04.6 glibc/2.23
                UID:GID : 1000:1000
             netrc file : None
           offline mode : False
```

**Comment by peastman** [comment]
Ok, it appears that if I use

```
conda install -c omnia-dev/label/beta openmm
```

it fails, but if I use

```
conda install -c omnia-dev/label/beta -c conda-forge openmm
```

it succeeds.  This build must have added a dependency to something on conda-forge?  The error message is very confusing, but it does work once you add that channel.

**Comment by peastman** [comment]
I don't have a Windows machine to test on, so I had to settle for a VM.  It has the same problem as Mac that it insists on installing a new Python.  Otherwise it works as far as I can tell.  `testInstallation` found the Reference, CPU, and OpenCL (using the CPU) platforms.  The CUDA plugins appear to be present, but of course I can't test using them.

**Comment by jchodera** [comment]
@peastman : What's in your `~/.condarc` file?

**Comment by peastman** [comment]
On Mac it's

```
channels:
  - omnia
  - conda-forge
  - defaults
```

On Linux that file doesn't exist.

**Comment by jaimergp** [comment]
We depend on several packages from conda-forge at build time and run time, most importantly `fftw`, which is ABI/compilers controlled [here](https://github.com/conda-forge/conda-forge-pinning-feedstock/blob/master/recipe/conda_build_config.yaml#L359). Conda Forge is required as a channel, as [per the docs](http://docs.openmm.org/latest/userguide/application.html#installing-openmm), and should have higher priority than `omnia`.

**Comment by Hong-Rui** [comment]
Hi, just an everyday OpenMM user wandering around

I suggest using this ~/.condarc file:

`channels:
 - conda-forge
 - omnia
 - defaults
channel_priority: disabled
show_channel_urls: true
`

after having this, do

`conda create -n openmm-beta python=3.7
conda activate openmm-beta
conda install openmm`

Don't use the -c flag since this would raise the specified channel on the top priority.
By setting channel_priority disabled, the newest package version number always go on first, and only if same version number appears the channel priority takes effects.

See also:
https://docs.conda.io/projects/conda/en/latest/user-guide/tasks/manage-channels.html

**Comment by jaimergp** [comment]
* `channel_priority` should be `strict` according to `conda-forge`
* For this particular beta, you need the channel`omnia-dev/label/beta`

**Comment by peastman** [comment]
Shall I go ahead and post an announcement?  For an actual release, having it replace the Python would be a serious problem.  But this is just a beta, and since we expect to build the release with a different infrastructure, this probably won't be relevant to it anyway.

**Comment by jchodera** [comment]
On my `osx` installation, creating a new environment without specifying the python version attempts to create a python 3.8 environment. Instead, I was able to specify 
```bash
conda create -n openmm-7.5.0-beta -c conda-forge -c omnia-dev/label/beta python=3.7 openmm 
```
which worked just fine.

Also, this installed the beta in an environment with python 3.7 without upgrading python:
```bash
conda install -c conda-forge -c omnia-dev/label/beta openmm=7.5.0
```
Maybe we can just be careful about recommending how people specify creation of an environment or updating of OpenMM?

**Comment by peastman** [comment]
What does `conda list | grep python` show in your base environment?  Possibly you've already replaced the default python with a conda-forge version, and that's why it isn't trying to replace it again?

**Comment by jchodera** [comment]
> What does conda list | grep python show in your base environment? Possibly you've already replaced the default python with a conda-forge version, and that's why it isn't trying to replace it again?

It's conda-forge python:
```
python                    3.7.6           cpython_h1fd5dd1_6    conda-forge
```
Don't forget that the omnia* instructions _require_ that you update your Python after adding the conda-forge channel with `conda update --all`. 

*now broken because we can't find who owns the DNS entry, or it has already lapsed

**Comment by Hong-Rui** [comment]
If you want to use the python from `defaults` channel, that means what you want is that the `defaults` channel has higher priority, and you want to use the python comes from `defaults` even if its version is older than that is in `conda-forge`.

Then what you should do is just
`conda install -c defaults -c conda-forge -c omnia-dev/label/beta openmm`
with `channel_priority: strict`.

Which is equivalent by setting a `~/.condarc` file stating

```
channels:
 - defaults
 - conda-forge
 - omnia-dev/label/beta

channel_priority: strict
show_channel_urls: true
```

and do `conda install openmm`

In any way, one should always figure out with an abstract rule on how you want the conda solver find the packages and all the dependencies for you, and this would correspond to different solver strategies, such as `strict`, `flexible`, and `disabled`.

```
channel_priority (ChannelPriority)
Accepts values of 'strict', 'flexible', and 'disabled'. The default
value is 'flexible'. With strict channel priority, packages in lower
priority channels are not considered if a package with the same name
appears in a higher priority channel. With flexible channel priority,
the solver may reach into lower priority channels to fulfill
dependencies, rather than raising an unsatisfiable error. With channel
priority disabled, package version takes precedence, and the
configured priority of channels is used only to break ties. In
previous versions of conda, this parameter was configured as either
True or False. True is now an alias to 'flexible'.

channel_priority: flexible
```

https://docs.conda.io/projects/conda/en/latest/user-guide/tasks/manage-channels.html


For my own personal perference, I would use always the newest version of packages from whatever the channels it comes from. Only with the same package version number then the channel priority plays a role. This correspond to `channel_priority: disabled`.

On the other hand, if my preference is always use the packages from the higher priority channels, even if the version number is older than the one from the low priority channels, then I need to choose `channel_priority: strict`.

If I want something compromised in between of these, then this corresponds to `channel_priority: flexible` (which is the default for current version of conda). However, notice that since this is an multi-target optimization problem, the conda solver may give you different results when you try it second times.

**Comment by peastman** [comment]
My view on it is much simpler: installing OpenMM should never change what Python is installed.  If I've chosen to install Python 3.7, regardless of what channel it came from, `conda install openmm` should download the Python 3.7 version of OpenMM and leave Python untouched.  It should never replace it with 3.8.

**Comment by Hong-Rui** [comment]
This does not depends on how you build the package of OpenMM, but depends on how users install it.
If a user creates an conda env saying `conda create -n openmm-beta python=3.7`, then he would never get an python 3.8 even if he do `conda update --all`, since by specifying `python=3.7` when creating env, conda solver will do an constraint optimization for all installation happens in this env.

If a user does not specify `python=3.7`, then he would have problem on seeing higher version of python coming from all his installation actions.

**Comment by jchodera** [comment]
@peastman: These are `conda` command issues, not OpenMM package issues. I'm in full agreement with @Hong-Rui (thanks, BTW!) that we just want to be explicit about instructions and provide a few examples for what users should do to achieve different desired outcomes.

**Comment by peastman** [comment]
I posted the announcement: https://simtk.org/plugins/phpBB/viewtopicPhpbb.php?f=161&t=12332&p=0&start=0&view=&sid=8fd4ba44bb7215e8496d67c6eee6ae50

**Comment by peastman** [comment]
Now that the beta is out, this is the time for final bug fixing and documentation updates.  I'm currently going through the issue tracker looking for things to fix.  If there are any you particularly want to see fixed, please post them here.

**Comment by jchodera** [comment]
@peastman : Can you tag everything that might be a bug as such with labels? That will make it easier to do this exercise in future!

**Comment by jchodera** [comment]
If we want to update the benchmarks for 7.5.0, let's also update the benchmarking script prior to release:
https://github.com/openmm/openmm/issues/2681

**Comment by peastman** [comment]
I already updated the benchmarking script a while ago.  I left that issue open because it also includes suggestions about creating a programmatic benchmarking API.

Anything that's been confirmed to be a bug should already be tagged appropriately, though I can't guarantee some haven't been missed.

**Comment by jchodera** [comment]
> I already updated the benchmarking script a while ago. I left that issue open because it also includes suggestions about creating a programmatic benchmarking API.

Thanks! I've updated the issue title to reflect this. (Feel free to do so as appropriate as well!)

> Anything that's been confirmed to be a bug should already be tagged appropriately, though I can't guarantee some haven't been missed.

I'd love to get to the bottom of any suspicious AMD issues like #2813. There's a large number of these on FAH we don't yet have insight into, but it will likely also take a few weeks before we can deploy an instrumented core that will run unit tests on failures, so we may not want to hold up 7.5.0 for that.

**Comment by Hong-Rui** [comment]
Hi, I remember there was an issue regarding the HMR exclusive for solute molecules but not on solvents. #2442
I walked through the codes and found that this has not been implemented. Have you postponed this to later release?

Recently we need this option and found that this haven't been implemented :(

btw, I also found things interesting that the ParmEd command line tool bundled with AmberTools actually has this built-in option on the ``HMassRepartition`` command and is default set to ``on`` by AMBER. 
However, the ParmEd python API does not have this option in ``Structure.createSystem``, same as OpenMM's ``Forcefield.createSystem``.

Just curious do you have plan on this feature?
Thanks.

**Comment by peastman** [comment]
I'd also like to figure out what's going on with AMD.  Unfortunately the only AMD GPU I have access to at the moment is the one in my laptop, and it doesn't have enough memory for large systems.

The HMR feature was postponed to a later release.  Sorry. :(  On the other hand, an easy workaround is to fix up the masses after calling `createSystem()`.  Something like this should work:

```python
for i,atom in enumerate(topology.atoms()):
  if atom.residue.name == 'HOH':
    system.setParticleMass(i, atom.element.mass)
```

**Comment by jchodera** [comment]
BTW, I count only six downloads of the new OpenMM 7.5.0 beta:
https://anaconda.org/omnia-dev/openmm/files?channel=beta

Part of the reason may be that we pushed it to the `omnia-dev` channel, while folks (and our travis infrastructures) may have expected it in the `omnia` channel. Should we copy or move it over to `omnia`? And try to reach more users?

Do we still have a mailing list? I set up a twitter account, but we only have ~500 people following that right now:
https://twitter.com/openmm_toolkit
